syntax = "proto3";

package auth;

service AuthService {
  rpc Register(RegisterRequest) returns (AuthResponse);
  rpc Login(LoginRequest) returns (AuthResponse);
  rpc ValidateToken(ValidateRequest) returns (ValidateResponse);
  rpc GetUserProfile(GetUserRequest) returns (UserProfile);
}

message RegisterRequest {
  string email = 1;
  string phone_number = 2;
  string password = 3;
  string role = 4;
  string bank_account = 5;
}

message LoginRequest {
  string email = 1;
  string password = 2;
}

message AuthResponse {
  string token = 1;
  string user_id = 2;
  string email = 3;
  string role = 4;
}

message ValidateRequest {
  string token = 1;
}

message ValidateResponse {
  bool valid = 1;
  string user_id = 2;
  string role = 3;
}

message GetUserRequest {
  string user_id = 1;
}

message UserProfile {
  string id = 1;
  string email = 2;
  string phone_number = 3;
  string role = 4;
  string bank_account = 5;
  string created_at = 6;
}